winpr/libwinpr/ncrypt/ncrypt.c

```diff
@@ -126,9 +126,18 @@ SECURITY_STATUS NCryptOpenStorageProvider(NCRYPT_PROV_HANDLE* phProvider, LPCWST
 {
 
 #ifdef WITH_PKCS11
-	if (_wcscmp(pszProviderName, MS_SMART_CARD_KEY_STORAGE_PROVIDER) == 0)
+	if (_wcscmp(pszProviderName, MS_SMART_CARD_KEY_STORAGE_PROVIDER) == 0 ||
+		_wcscmp(pszProviderName, MS_SCARD_PROV) == 0)
 	{
-		static LPCSTR openscPaths[] = { "/usr/lib/x86_64-linux-gnu/pkcs11/opensc-pkcs11.so", NULL };
+		static LPCSTR openscPaths[] = {
+#ifdef __APPLE__
+				"/usr/local/lib/pkcs11/opensc-pkcs11.so",
+#else
+				/* linux and UNIXes */
+				"/usr/lib/x86_64-linux-gnu/pkcs11/opensc-pkcs11.so",
```

Comment:
this looks wrong.
1. There are still 32bit builds out there
2. There are these that use the `lib64` style paths (e.g. fedora)

Reply:
Partially fixed (some paths added should cover most cases).